Core/EventHandlers.lua

```diff
@@ -1325,12 +1325,14 @@ function R:OnEvent(event, ...)
 			end
 		end
 
-		-- Handle opening Forgotten Chest (Venthyr only chest for Silessa's Battle Harness in Revendreth, Shadowlands)
-		if Rarity.isFishing and Rarity.isOpening and Rarity.lastNode and (Rarity.lastNode == L["Forgotten Chest"]) then
-			local names = {"Silessa's Battle Harness"}
+		-- Handle opening Forgotten Chest (Shadowlands, Revendreth chest for Stony's Infused Ruby pet and Silessa's Battle Harness mount)
+		if Rarity.isFishing and Rarity.isOpening and Rarity.lastNode and (Rarity.lastNode == L["Forgotten Chest"])
+			and GetBestMapForUnit("player") ~= CONSTANTS.UIMAPIDS.STORMSONG_VALLEY -- Chest with the same name in Stormsong Valley
+		then
+			local names = {"Stony's Infused Ruby", "Silessa's Battle Harness"}
 			Rarity:Debug("Detected Opening on " .. L["Forgotten Chest"] .. " (method = SPECIAL)")
 			for _, name in pairs(names) do
-				local v = self.db.profile.groups.items[name] or self.db.profile.groups.mounts[name]
+				local v = self.db.profile.groups.pets[name] or self.db.profile.groups.mounts[name]
```

Comment:
I see you replaced ``items`` with ``pets``. I assume this was an unrelated correction and that the code was wrong before, since the item should always have been in the ``pets`` category?

Reply:
It seems this is a symptom of copy and pasting the logic from other containers/chests. I think it isn't necessary on a bunch of them. 

I replaced `items` for `pets` in this one, as I combined the pet with the mount in one if statement, rather than have the duplicated logic we had before.

Reply:
Oh, gotcha. Yeah, this whole segment needs streamlining, but that's for another time :)